### PR TITLE
Look up prefixed variables first

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

**Comment by pakrym** [comment]
test

File: sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs [hunk 1]
```diff
@@ -229,15 +229,19 @@ protected string GetOptionalVariable(string name)
...
+            var value = Environment.GetEnvironmentVariable(prefixedName);
 
             if (value == null)
             {
                 _environmentFile.TryGetValue(prefixedName, out value);
             }
 
+            if (value == null)
+            {
+                Environment.GetEnvironmentVariable(name);
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs]: lol..

File: sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs [hunk 2]
```diff
@@ -19,13 +19,13 @@ namespace Azure.Core.TestFramework
...
     {
-        private static readonly string RepositoryRoot;
+        public static string RepositoryRoot { get; }
         private readonly string _prefix;
 
         private TokenCredential _credential;
         private TestRecording _recording;
 
-        private readonly Dictionary<string, string> _environmentFile = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        internal readonly Dictionary<string, string> _environmentFile = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:28]: Do you need this internal?

File: sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs [hunk 3]
```diff
@@ -19,13 +19,13 @@ namespace Azure.Core.TestFramework
     /// </summary>
     public abstract class TestEnvironment
     {
-        private static readonly string RepositoryRoot;
+        public static string RepositoryRoot { get; }
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:22]: Keep this internal.

File: sdk/core/Azure.Core/tests/TestEnvironmentTests.cs [hunk 4]
```diff
@@ -2,24 +2,47 @@
...
 
             Environment.SetEnvironmentVariable("CORE_Base64Secret", "1");
             Environment.SetEnvironmentVariable("CORE_CustomSecret", "1");
             Environment.SetEnvironmentVariable("CORE_DefaultSecret", "1");
             Environment.SetEnvironmentVariable("CORE_ConnectionStringWithSecret", "endpoint=1;key=2");
+            _envFilePath = Path.Combine(TestEnvironment.RepositoryRoot, "sdk", "core", "test-resources.json.env");
+            using FileStream stream = File.Create(_envFilePath);
+            Dictionary<string, string> envFile = new Dictionary<string, string>();
+            envFile.Add("CORE_TENANT_ID", "7");
+            var serializer = new JsonObjectSerializer();
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/TestEnvironmentTests.cs:36]: Do you use this?

File: sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:22]: It's needed from Azure.Core.Tests. We don't have IVT enabled.

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:22]: Will enable IVT.

Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs:22]: Tried enabling but there are conflicts between shared types from Azure.Core and TestFramework.

**Review by pakrym — APPROVED** [review]

File: sdk/core/Azure.Core/tests/TestEnvironmentTests.cs [hunk 5]
```diff
@@ -2,24 +2,59 @@
...
             Environment.SetEnvironmentVariable("CORE_DefaultSecret", "1");
             Environment.SetEnvironmentVariable("CORE_ConnectionStringWithSecret", "endpoint=1;key=2");
+
+            // Env file is only supported on Windows.
+            if (s_isWindows)
+            {
+                _envFilePath = Path.Combine(TestEnvironment.RepositoryRoot, "sdk", "core", "test-resources.json.env");
+                using FileStream stream = File.Create(_envFilePath);
+                Dictionary<string, string> envFile = new Dictionary<string, string>
+            {
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/TestEnvironmentTests.cs:42]: nit: format

**Pull request merged**